questionGenerator.py

```diff
@@ -469,34 +396,57 @@ def modifyDuplicateTopics(topicList):
     return topicList
 
 
-def getDominantTopic(clusteredTopics):
+def getRelevantRegions(clusteredTopics, questionCount=-1, minFrequency=2):
     """
-    Returns the dominant topics based on the frequency of occurrence.
+    Retrieves relevant text regions based on the given parameters.
 
-    Parameters:
-    clusteredTopics (DataFrame): A DataFrame containing clustered topics with columns "Topic" and "Frequency".
+    Args:
+        clusteredTopics (DataFrame): A DataFrame containing clustered topics and their frequencies.
+        questionCount (int, optional): The number of questions to generate. Defaults to -1.
+        minFrequency (int, optional): The minimum frequency of a topic to be considered relevant. Defaults to 2.
 
     Returns:
-    dominantTopics (DataFrame): A DataFrame containing the dominant topics sorted by start time.
+        DataFrame: A DataFrame containing the relevant text regions.
+
+    Raises:
+        SystemExit: If no relevant text regions are found.
+
     """
-    dominantTopics = []
-    for group in clusteredTopics.groupby("Topic"):
-        dominantTopics.append(
-            group[1].sort_values("Frequency", ascending=False).head(1)
+    if questionCount == -1:
+        # Retrieve one question per topic, by finding the text with the highest occurence of that topic.
+        logging.info(f"No question count set. Retrieving one question per topic.")
+        relevantRegions = []
+        for group in clusteredTopics.groupby("Topic"):
+            relevantRegions.append(
+                group[1].sort_values("Frequency", ascending=False).head(1)
+            )
+        relevantRegions = pd.concat(relevantRegions).sort_values("Start")
+
+    else:
+        # Retrieve the relevant text regions sorted by frequency of topic occurence.
+        logging.info(f"Identifying relevant text for {questionCount} questions.")
+        sortedTopics = clusteredTopics.sort_values(
+            ["Frequency", "Topic"], ascending=[False, True]
         )
+        sortedTopics = sortedTopics[sortedTopics["Frequency"] >= minFrequency]
 
-    dominantTopics = pd.concat(dominantTopics).sort_values("Start")
+        if questionCount > sortedTopics.shape[0]:
+            logging.info(
+                f"Question count is greater than the number of regions with relevant text. 
+                {sortedTopics.shape[0]} questions will be generated instead."
+            )
```

Comment:
I see the below syntax error 
```
Traceback (most recent call last):
  File "/Users/pushyami/git_tl_projects/annoto-gai/captionsProcessor.py", line 4, in <module>
    from questionGenerator import retrieveQuestions
  File "/Users/pushyami/git_tl_projects/annoto-gai/questionGenerator.py", line 435
    f"Question count is greater than the number of regions with relevant text.
    ^
SyntaxError: unterminated string literal (detected at line 435)
```

Reply:
I have fixed this in the latest commit. I don't know why my VSCode IDE or the formatter did not recognize this as an error.